akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala

```diff
@@ -257,14 +259,19 @@ class FlowSplitWhenSpec extends StreamSpec {
     }
 
     "fail substream if materialized twice" in assertAllStagesStopped {
+      implicit val mat = ActorMaterializer(ActorMaterializerSettings(system)
+        .withInputBuffer(initialSize = 1, maxSize = 1))
+
       import system.dispatcher
+      val probe = Source(1 to 5).splitWhen(_ ⇒ true).lift
+        .map { src ⇒ src.runWith(Sink.ignore)(mat).flatMap(_ ⇒ src.runWith(Sink.ignore)(mat)) }
+        .runWith(TestSink.probe[Future[Done]])(mat)
+      probe.request(1)
+      val future = probe.requestNext()
       an[IllegalStateException] mustBe thrownBy {
-        Await.result(
-          Source.single(1).splitWhen(_ ⇒ true).lift
-            .mapAsync(1) { src ⇒ src.runWith(Sink.ignore).flatMap(_ ⇒ src.runWith(Sink.ignore)) } // Sink.ignore+mapAsync pipes error back
-            .runWith(Sink.ignore),
-          3.seconds)
+        Await.result(future, 3.seconds)
       }
```

Comment:
I could not reproduce it at the first place. However, I changed mapAsync to map and test failed. The reason was that stream is finished at a time when the second materialization failed substream so no exception was propagated. My best guess was that there is no issue with code (until you think differently) but with test itself - so I rewrote it to have the stream wait.